homeassistant/components/tuya/climate.py

```diff
@@ -420,7 +424,7 @@ def hvac_mode(self) -> HVACMode:
         if not self.device.status.get(DPCode.SWITCH, True):
             return HVACMode.OFF
 
-        if DPCode.MODE not in self.device.function:
+        if (self._is_thermostat) or (DPCode.MODE not in self.device.function):
```

Comment:
Basically, this is only one place left that needs to be improved. With my changes it react only to switch status and ignore modes. Seems that this is incorrect in general, but on the other hand, without this it doesn't work in my particular case.

Reply:
With changes in modes list creation, this change is not needed anymore